client/tests/webdriver/baseSpecs/mergePeople.spec.js

```diff
@@ -389,3 +410,154 @@ describe("Merge people page", () => {
     )
   })
 })
+
+describe("Merge people of different roles", () => {
+  it("Should select a principal for the left side", () => {
+    // Open merge people page.
+    MergePeople.open()
+    MergePeople.title.waitForExist()
+    MergePeople.title.waitForDisplayed()
+    // Search and select a person from left person field.
+    MergePeople.leftPersonField.setValue(EXAMPLE_PEOPLE.validLeft.search)
+    MergePeople.waitForAdvancedSelectLoading(EXAMPLE_PEOPLE.validLeft.fullName)
+    MergePeople.firstItemFromAdvancedSelect.click()
+    // Check if the fields displayed properly after selecting a person from left side.
+    MergePeople.waitForColumnToChange(
+      EXAMPLE_PEOPLE.validLeft.name,
+      "left",
+      "Name"
+    )
+  })
+  it("Sould select an advisor for the right side", () => {
+    MergePeople.rightPersonField.setValue(EXAMPLE_PEOPLE.advisorRight.search)
+    MergePeople.waitForAdvancedSelectLoading(
+      EXAMPLE_PEOPLE.advisorRight.fullName
+    )
+    MergePeople.firstItemFromAdvancedSelect.click()
+    // Check if the fields displayed properly after selecting a person from left side.
+    MergePeople.waitForColumnToChange(
+      EXAMPLE_PEOPLE.advisorRight.name,
+      "right",
+      "Name"
+    )
+  })
+  it("Should display the different roles warning", () => {
+    MergePeople.waitForDifferentRolesAlert()
+  })
+  it("Should not display the select buttons on each side", () => {
+    MergePeople.getSelectButton("left", "Name").waitForDisplayed({
+      reverse: true
+    })
+    MergePeople.getSelectButton("right", "Name").waitForDisplayed({
+      reverse: true
+    })
```

Comment:
Perhaps also assert that you can't Edit History Manually in this case, and that the "Clear field value" buttons are unavailable?